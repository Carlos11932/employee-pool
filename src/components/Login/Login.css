
.login-container {
    max-width: 400px;
    margin: 20px auto;
    padding: 20px;
    background: white;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  }
  
  .login-container h2 {
    text-align: center;
  }
  
  .login-container form {
    display: flex;
    flex-direction: column;
  }
  
  .login-container form select,
  .login-container form button {
    margin-top: 10px;
  }