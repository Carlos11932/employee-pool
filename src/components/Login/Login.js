import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { setAuthedUser } from '../../actions/authedUser';
import './Login.css';

const Login = () => {
  const [user, setUser] = useState('');
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const users = useSelector(state => state.users);

  const handleChange = (e) => {
    setUser(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (user) {
      dispatch(setAuthedUser(user));
      const lastRequestedPage = localStorage.getItem('lastRequestedPage');
      if (lastRequestedPage) {
        navigate(lastRequestedPage);
        localStorage.removeItem('lastRequestedPage');
      } else {
        navigate('/');
      }
    }
  };

  return (
    <div className="login-container">
      <h2>Login</h2>
      <form onSubmit={handleSubmit}>
        <select value={user} onChange={handleChange}>
          <option value="">Select a user</option>
          {Object.keys(users).map(userId => (
            <option key={userId} value={userId}>
              {users[userId].name}
            </option>
          ))}
        </select>
        <button type="submit">Login</button>
      </form>
    </div>
  );
};

export default Login;