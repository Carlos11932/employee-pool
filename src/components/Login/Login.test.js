import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Login from './Login';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Login', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      users: {
        sarahedo: {
          id: 'sarahedo',
          name: 'Sarah Edo',
          avatarURL: 'Sarah_Edo.webp',
        },
        tylermcginnis: {
          id: 'tylermcginnis',
          name: 'Tyler McGinnis',
          avatarURL: 'Tyler_McGinnis.webp',
        },
      },
    });
  });

  it('should render login component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});