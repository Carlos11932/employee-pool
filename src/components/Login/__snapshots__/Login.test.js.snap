// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="login-container"
  >
    <h2>
      Login
    </h2>
    <form>
      <select>
        <option
          value=""
        >
          Select a user
        </option>
        <option
          value="sarahedo"
        >
          Sarah Edo
        </option>
        <option
          value="tylermcginnis"
        >
          Tyler McGinnis
        </option>
      </select>
      <button
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Login should render login component 1`] = `
<DocumentFragment>
  <div
    class="login-container"
  >
    <h2>
      Login
    </h2>
    <form>
      <select>
        <option
          value=""
        >
          Select a user
        </option>
        <option
          value="sarahedo"
        >
          Sarah Edo
        </option>
        <option
          value="tylermcginnis"
        >
          Tyler McGinnis
        </option>
      </select>
      <button
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</DocumentFragment>
`;
